### ISSUE TITLE
Debug Levels?

### ISSUE THREAD

**Issue by NHDaly** [MasonProtter/ToggleableAsserts.jl#2]
I've been wanting to switch our codebase at RelationalAI to use this! :)

One thing that we have that's not yet covered by this package is *debug levels* in our assertions. For example:

```julia
@dassert1 false  # Runs if DEBUG_LEVEL >= 1
@dassert2 false  # Runs if DEBUG_LEVEL >= 2
...
```

I've been wanting to rewrite this a bit to put the number into a separate argument, e.g.:

```julia
@dassert 1 false  # Runs if DEBUG_LEVEL >= 1
@dassert 2 false  # Runs if DEBUG_LEVEL >= 2
```

So then, if we were to use this package, I think instead of having a `toggle_asserts(onoff::Bool)` that we would use, we'd probably rather have a `set_debug_level(level::Int)`, which would `@eval debug_level() = $level`.

Should we create a separate package for this? Or roll this into this package?

Thanks! :) And thanks for pointing us to this idea, @MasonProtter! I love this idea, and I can't believe we never thought of it!

**Comment by NHDaly** [comment]
CC: @sacha0, @li1

**Comment by NHDaly** [comment]
CC: @li1

**Comment by MasonProtter** [comment]
Yeah, I think that’s a great idea and I’d be happy to review and collaborate on a PR for it!

For backwards comparability, I suppose we could treat `true` as debug level `1` (Or inf?) and `false` as zero?

**Comment by oskarr** [comment]
I threw together something for this quickly in a [fork](https://github.com/oskarr/ToggleableAsserts.jl). Just needed something that works for now, and haven't really thought a lot about the public API and potential deprecations, etc. I'm pretty sure it's backwards-compatible. It passes the old tests, but I haven't tested it more extensively than that. Feel free to pull or modify it if it looks "good enough" (if there's still any interest in this).
